firedrake/functionspaceimpl.py

```diff
@@ -493,29 +493,38 @@ def exterior_facet_boundary_node_map(self, method):
         """
         return self._shared_data.exterior_facet_boundary_node_map(self, method)
 
+    def boundary_nodes(self, sub_domain, method):
+        """Return the boundary nodes for this :class:`~.FunctionSpace`.
+
+        :arg sub_domain: the mesh marker selecting which subset of facets to consider.
+        :arg method: the method for determining boundary nodes.
+        :returns: A numpy array of the unique function space nodes on
+           the selected portion of the boundary.
+
+        See also :class:`~.DirichletBC` for details of the arguments.
+        """
+        return self._shared_data.boundary_nodes(self, sub_domain, method)
```

Comment:
Probably not necessary, should splat `top_bottom_...` into `boundary_nodes` and remove other bits.